lib/esestream.cpp

```diff
@@ -59,11 +53,27 @@ ESEStream::~ESEStream ()
       m_reader.fileSize ());
 }
 
+void
+ESEStream::reset()
+{
+  m_eos = false;
+  m_bufferPosition = 0;
+  m_frameCount = 0;
+  m_nextPacket = nullptr;
```

Comment:
Will this not leak a packet if there was one pending? Not sure if we're running Valgrind in the test suite yet.